tools/perf_test_cgal.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Emmett Lalish & contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_mesh_processing/corefinement.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+
+#include "manifold.h"
+
+using namespace manifold;
+
+// Epick = Exact predicates Inexact constructions. Seems fair to use to compare
+// to Manifold, which seems to store double coordinates.
+typedef CGAL::Epick Kernel;
+
+// Epeck = Exact predicates Exact constructions. What OpenSCAD uses to guarantee
+// geometry ends up where it should even after many operations. typedef
+// CGAL::Epeck Kernel;
+
+typedef CGAL::Point_3<Kernel> Point;
+typedef CGAL::Surface_mesh<Point> TriangleMesh;
+
+void manifoldToCGALSurfaceMesh(Manifold &manifold, TriangleMesh &cgalMesh) {
+  typedef boost::graph_traits<TriangleMesh> GT;
+  typedef typename GT::vertex_descriptor vertex_descriptor;
+
+  auto maniMesh = manifold.GetMesh();
+
+  std::unordered_map<size_t, vertex_descriptor> vertices;
+  for (size_t i = 0, n = maniMesh.vertPos.size(); i < n; i++) {
+    auto &vert = maniMesh.vertPos[i];
+    vertices[i] = cgalMesh.add_vertex(Point(vert.x, vert.y, vert.z));
+  }
+
+  for (auto &triVert : maniMesh.triVerts) {
+    std::vector<vertex_descriptor> polygon{
+        vertices[triVert[0]], vertices[triVert[1]], vertices[triVert[2]]};
+    cgalMesh.add_face(polygon);
+  }
+}
+
+int main(int argc, char **argv) {
+  for (int i = 0; i < 8; ++i) {
+    Manifold sphere = Manifold::Sphere(1, (8 << i) * 4);
+    Manifold sphere2 = sphere;
+    sphere2.Translate(glm::vec3(0.5));
+
+    TriangleMesh cgalSphere, cgalSphere2, cgalOut;
+    manifoldToCGALSurfaceMesh(sphere, cgalSphere);
+    manifoldToCGALSurfaceMesh(sphere2, cgalSphere2);
+
+    auto start = std::chrono::high_resolution_clock::now();
+    auto result =
```

Comment:
Is the result also a triangle mesh, or is it a polygon mesh?

Reply:
Yes [corefinement](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corefinement__grp.html) consumes and produces triangles meshes, even though Surface_mesh is more generic.